test/per_file_coverage.sh

```diff
@@ -21,7 +21,7 @@ declare -a KNOWN_LOW_COVERAGE=(
 "source/common/tcp:94.6"
 "source/common/thread:0.0" # Death tests don't report LCOV
 "source/common/watchdog:58.6" # Death tests don't report LCOV
-"source/exe:91.5"
+"source/exe:78.8" # Binary with extensions fails to link due to the size, and main() lacks coverage.
```

Comment:
Looking at
https://storage.googleapis.com/envoy-postsubmit/f313fc6/coverage/source/exe/main_common.cc.gcov.html
vs
https://storage.googleapis.com/envoy-pr/df2d030/coverage/source/exe/main_common.cc.gcov.html 
it looks like most of the drop is no longer covering hot restart in the coverage build.  Why can't we fix that?

Reply:
the size of the binary seems to have grown - this is throwing 2 compile errors - the first of which was resovled here https://github.com/envoyproxy/envoy/blob/6cc85934c84f4f2a485ba13fdc0e8e625ee01a00/.bazelrc#L229-L230

this PR then through another related error - the fixes/workaround for that error would appear to have more impact

Reply:
yep, I think we need to look into the drop rather than just landing as-is.

Reply:
related https://github.com/envoyproxy/envoy/pull/31655#discussion_r1443614654

Reply:
I'm not sure I see that the hot restart code is affected. The difference is the literal `main()` function try-catch block, because that requires a full binary to be run via a shell script

Reply:
Ok we can instead add a shell script which doesn't require all the extensions, no?

Reply:
Yes, possibly, but that requires figuring out which test covered which line in the previous coverage and replicating it with a new smaller test. We can certainly do it, but it'd push this PR to have more stuff unrelated to CEL.

Reply:
Yeah but folks follow-up with removing coverage is pretty grim.  If you want to land it as a separate PR ahead of this one feel free, but I don't think we should lower coverage to bump deps.